src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,50 @@
+package mate.academy.dao;
+
+import mate.academy.Exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import java.util.Optional;
+
+@Dao
+public class MovieDaoImpl implements MovieDao{
+    @Override
+    public Movie add(Movie movie) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
```

Comment:
It is better to make it private static field of a class, and use it instead of creating it here and in get() method

Reply:
Done